### DISCUSSION TITLE
dpgen init_ bulk method help

### DISCUSSION THREAD

**Discussion by LJyanyan** [deepmodeling/dpgen#1036] [Q&A]
Hi, I'm using dpgen to run LiSi. I want to use init_ bulk method generates the initial dataset, and then always reports an error in the invalid poscar format; At the same time, I also tried the example of CH4. The original files were copied from the example folder, and the problem was also reported. I would be grateful if someone could help me.
**LiSi:**
Description
------------
Traceback (most recent call last):
  File "/home/deepmd/.local/bin/dpgen", line 8, in <module>
    sys.exit(main())
  File "/home/deepmd/.local/lib/python3.10/site-packages/dpgen/main.py", line 185, in main
    args.func(args)
  File "/home/deepmd/.local/lib/python3.10/site-packages/dpgen/data/gen.py", line 1410, in gen_init_bulk
    make_super_cell_poscar(jdata)
  File "/home/deepmd/.local/lib/python3.10/site-packages/dpgen/data/gen.py", line 335, in make_super_cell_poscar
    from_struct.to('poscar',to_file)  
  File "/home/deepmd/.local/lib/python3.10/site-packages/pymatgen/core/structure.py", line 2518, in to
    raise ValueError(f"Invalid format: `{str(fmt)}`")
ValueError: Invalid format: `poscar_lisi.01x01x01/00.place_ele/poscar`
**CH4**
Description
------------
Traceback (most recent call last):
  File "/home/deepmd/.local/bin/dpgen", line 8, in <module>
    sys.exit(main())
  File "/home/deepmd/.local/lib/python3.10/site-packages/dpgen/main.py", line 185, in main
    args.func(args)
  File "/home/deepmd/.local/lib/python3.10/site-packages/dpgen/data/gen.py", line 1410, in gen_init_bulk
    make_super_cell_poscar(jdata)
  File "/home/deepmd/.local/lib/python3.10/site-packages/dpgen/data/gen.py", line 335, in make_super_cell_poscar
    from_struct.to('poscar',to_file)  
  File "/home/deepmd/.local/lib/python3.10/site-packages/pymatgen/core/structure.py", line 2518, in to
    raise ValueError(f"Invalid format: `{str(fmt)}`")
ValueError: Invalid format: `ch4.poscar.01x01x01/00.place_ele/poscar`

**Comment by HuangJiameng** [comment]
could you provide the POSCAR file mentioned above?

**Reply by LJyanyan** [reply]
**LiSi**
Li16Si16
 1.0
 9.327158 0.000000 0.000000
 0.000000 9.327158 0.000000
 0.000000 0.000000 5.740710
  Li   Si
  16   16
Direct
 0.580968 0.135521 0.929329
 0.864479 0.080968 0.679329
 0.135521 0.919032 0.679329
 0.419032 0.864479 0.929329
 0.419032 0.364478 0.820671
 0.635521 0.919032 0.570671
 0.364478 0.080968 0.570671
 0.580968 0.635521 0.820671
 0.080968 0.635521 0.429329
 0.364478 0.580968 0.179329
 0.635521 0.419032 0.179329
 0.919032 0.364478 0.429329
 0.919032 0.864479 0.320671
 0.135521 0.419032 0.070671
 0.864479 0.580968 0.070671
 0.080968 0.135521 0.320671
 0.610680 0.202658 0.469622
 0.797342 0.110680 0.219622
 0.202658 0.889320 0.219622
 0.389321 0.797342 0.469622
 0.389321 0.297341 0.280378
 0.702658 0.889320 0.030378
 0.297341 0.110680 0.030378
 0.610680 0.702658 0.280378
 0.110680 0.702658 0.969623
 0.297341 0.610680 0.719623
 0.702658 0.389321 0.719623
 0.889320 0.297341 0.969623
 0.889320 0.797342 0.780377
 0.202658 0.389321 0.530377
 0.797342 0.610680 0.530377
 0.110680 0.202658 0.780377
**CH4**
POSCAR file written by OVITO
   1.0000000000000000
    10.0000000000000000    0.0000000000000000    0.0000000000000000
     0.0000000000000000   10.0000000000000000    0.0000000000000000
     0.0000000000000000    0.0000000000000000   10.0000000000000000
   H    C
     4     1
Direct
  0.5381543385532134  0.4068608007477559  0.3605730142093164
  0.3945396557300829  0.4803205689390709  0.4384688400135213
  0.5520924283424902  0.5654502852627673  0.4427087377775140
  0.5281853047372259  0.4164147551778533  0.5391826568442692
  0.5032505926369848  0.4672551598725561  0.4452323411553799

  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
Both poscars are provided, thank you! Where might my problem arise?

**Comment by HuangJiameng** [comment]
When I try to read the POSCAR with dpdata (which DP-GEN read this file with), the following error occurs.
Could you check the version of vasp in your environment?
<img width="923" alt="image" src="https://user-images.githubusercontent.com/105633685/202615926-b339d32e-1bd4-4fb2-a2c2-699ab396bb43.png">

**Reply by LJyanyan** [reply]
The vasp version on the server is vasp. 5.4. 
I raised this problem in the DP user Q&A group. The problem has been solved because of the pymatgen version. I said that the latest version of dpgen is not compatible with pymatgen, and that you can replace pymatgen==2022.0.17. Thank you again for your help. I wish you success in your work and a happy life.

**Comment by JTaozhang** [comment]
Do you find the solution of ValueError: Invalid format: ch4.poscar.01x01x01/00.place_ele/poscar? I also meet this problem? if you could provide some guidance, I would greatly appreciate your help.

**Comment by LJyanyan** [comment]
You can refer to this tutorial, URL: https://github.com/deepmodeling/dpgen/issues/1108. I tried changing the pymatgen version to pymatgen==2022.0.17 and it worked.



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年3月18日(星期六) 下午5:55
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [deepmodeling/dpgen] dpgen init_ bulk method help (Discussion #1036)






你找到value的解决方案了吗错误:无效的格式:ch4.poscar.01x01x01/00.place_ele/poscar？我也遇到这个问题？如果您能提供一些指导，我将非常感谢您的帮助。

—
直接回复这封邮件，在GitHub上查看或取消订阅.
***@***.***与&gt;.

**Comment by JTaozhang** [comment]
Thank you, According to your suggestion, the problem solved (dpgen==0.11.0 and pymatgen==2022.0.17 wok well). This means that there still have compatibility problem between dpgen and pymatgen untill now, the latest version of pymatgen(2023.3.10) is incompatible with dpgen (dpgen==0.11.0), although the author said they have upgraded the code. Many thanks for your help again.
